server/mscalendar/availability.go

```diff
@@ -41,16 +43,17 @@ func (m *mscalendar) Sync(mattermostUserID string) (string, error) {
 func (m *mscalendar) SyncAll() (string, error) {
 	err := m.Filter(withSuperuserClient)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("not able to filter the super user client. %w", err)
 	}
 
 	userIndex, err := m.Store.LoadUserIndex()
 	if err != nil {
 		if err.Error() == "not found" {
 			return "No users found in user index", nil
 		}
-		return "", err
+		return "", fmt.Errorf("not able to load users from user index. %w", err)
 	}
+	m.Logger.Debugf("Users loaded successfully from user index")
```

Comment:
Since this will log on every run of this job, I think we should remove this log here. The improved verbosity of the error messages will work great as a substitute.